### PR TITLE
[KYUUBI #5944] Introduce Prometheus and Grafana for Kyuubi Playground

### LINKED ISSUE #5944: [Improvement] Introduce Prometheus and Grafana for Kyuubi Playground
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Improve Kyuubi playground module, help developer to build kyuubi dashboard.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zwangsheng**
# :mag: Description
## Issue References 🔗

This pull request fixes #5944

## Describe Your Solution 🔧

Add prometheus and grafana docker image to docker compose yaml, and by defaults, prometheus will get kyuubi server's metrics and grafana set prometheus as default datasource.

Help developers quickly build a dashboard development environment.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

In local test
- [x] all container start success
- [x] By defaults, prometheus get kyuubi's metrics
- [x] Grafana take the prometheus as default  datasource

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: docker/playground/README.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ Kyuubi supply some built-in dataset, after Kyuubi started, you can run the follo
 - MinIO: http://localhost:9001
 - PostgreSQL localhost:5432 (username: postgres, password: postgres)
 - Spark UI: http://localhost:4040 (available after Spark application launching by Kyuubi, port may be 4041, 4042... if you launch more than one Spark applications)
+- Prometheus: http://localhost:9090
+- grafana: http://localhost:3000 (username: admin, password: admin)
```
Comment (pan3793) [inline docker/playground/README.md:38]: grafana => Grafana 

I remember we can disable the authentication in conf

File: docker/playground/.env [hunk 2]
```diff
@@ -28,3 +28,5 @@ SPARK_VERSION=3.4.2
 SPARK_BINARY_VERSION=3.4
 SPARK_HADOOP_VERSION=3.3.4
 ZOOKEEPER_VERSION=3.6.3
+PROMETHEUS_VERSION=v2.45.2
```
Comment (pan3793) [inline docker/playground/.env:31]: why leave a leading `v`?

Comment (zwangsheng) [reply docker/playground/.env:31]: prometheus docker image requires.

See more tags in https://hub.docker.com/r/prom/prometheus/tags

Comment (pan3793) [reply docker/playground/.env:31]: that's tag, not version. if wanted, we may use `apache-kyuubi:${KYUUBI_VERSION}-spark` or something

File: docker/playground/README.md (hunk 1 above)
Comment (zwangsheng) [reply docker/playground/README.md:38]: I have tried to set anonymous access grafana, and actually here are same config to allow anonymous to create and edit dashboard, but it required login in to save those dashboard.

Comment (zwangsheng) [reply docker/playground/README.md:38]: After changing prometheus datasource not as default, anonymous can create those folders and dashboards.

Comment (pan3793) [reply docker/playground/README.md:38]: do you forget to update this line then?

File: docker/playground/prometheus/prometheus.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+scrape_configs:
+  - job_name: kyuubi
+    static_configs:
+    - targets: ['kyuubi:10019']
```
Comment (pan3793) [inline docker/playground/prometheus/prometheus.yml:21]: let's explicitly set the related configurations in `kyuubi-defaults.conf`, to make the user clear when he/she searches the codebase with 10019

**Review by pan3793 — CHANGES_REQUESTED** [review]
LGTM except to one line

File: docker/playground/README.md (hunk 1 above)
Comment (zwangsheng) [reply docker/playground/README.md:38]: let's keep username and password here, in case developer want to login in.

Comment (pan3793) [reply docker/playground/README.md:38]:
```suggestion
- Grafana: http://localhost:3000 (username: admin, password: admin)
```

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**